xdsl/frontend/onnx/ir_builder.py

```diff
@@ -1,16 +1,42 @@
-from onnx import ValueInfoProto
+from onnx import NodeProto, ValueInfoProto
 
+from xdsl.dialects import onnx
 from xdsl.frontend.onnx.type import get_type
-from xdsl.ir import Attribute
+from xdsl.ir import Attribute, SSAValue
+from xdsl.irdl import IRDLOperation
+
+OP_BY_OP_TYPE: dict[str, type[IRDLOperation]] = {
+    "Add": onnx.Add,
+    "Sub": onnx.Sub,
+}
+"""Associate the name of the operations with the respective operation in ONNX dialect."""
 
 
 class OnnxXdslMapping:
     """The representation of the onnx context."""
 
     type_by_name: dict[str, Attribute]
+    value_by_name: dict[str, SSAValue]
 
     def __init__(self):
         self.type_by_name = {}
+        self.value_by_name = {}
+
+
+def visit_node(node: NodeProto, ctx: OnnxXdslMapping) -> None:
+    """Update the onnx context with the current node of the onnx graph."""
+    if node.op_type not in OP_BY_OP_TYPE:
+        raise ValueError(f"Unknown ONNX op name {node.op_type}")
+
+    op = OP_BY_OP_TYPE[node.op_type]
+
+    operands = tuple(ctx.value_by_name[name] for name in node.input)
+    result_types = tuple(ctx.type_by_name[name] for name in node.output)
+
+    results = op.build(operands=operands, result_types=result_types).results
+
+    for output_name, result in zip(node.output, results, strict=True):
```

Comment:
What operations have multiple outputs? It feels like we might be better off raising `NotImplementedError` if the result count is anything but 1, that way we catch the case when it comes up.